hoss/coordinate_utilities.py

```diff
@@ -179,42 +182,234 @@ def get_1D_dim_array_data_from_dimvalues(
 
     dim_min = dim_values[0] - (dim_resolution * dim_indices[0])
     dim_max = dim_values[1] + (dim_resolution * (dim_size - 1 - dim_indices[1]))
+
     return np.linspace(dim_min, dim_max, dim_size)
 
 
 def get_valid_indices(
-    coordinate_row_col: np.ndarray, coordinate: VariableFromDmr
+    lat_lon_array: np.ndarray, coordinate: VariableFromDmr
 ) -> np.ndarray:
     """
     Returns indices of a valid array without fill values if the fill
     value is provided. If it is not provided, we check for valid values
     for latitude and longitude
     """
+
     # get_attribute_value returns a value of type `str`
     coordinate_fill = coordinate.get_attribute_value('_FillValue')
     if coordinate_fill is not None:
-        is_not_fill = ~np.isclose(coordinate_row_col, float(coordinate_fill))
+        is_not_fill = ~np.isclose(lat_lon_array, float(coordinate_fill))
     else:
         # Creates an entire array of `True` values.
-        is_not_fill = np.ones_like(coordinate_row_col, dtype=bool)
+        is_not_fill = np.ones_like(lat_lon_array, dtype=bool)
 
     if coordinate.is_longitude():
-        valid_indices = np.where(
-            np.logical_and(
-                is_not_fill,
-                np.logical_and(
-                    coordinate_row_col >= -180.0, coordinate_row_col <= 360.0
-                ),
-            )
-        )[0]
+        valid_indices = np.logical_and(
+            is_not_fill,
+            np.logical_and(lat_lon_array >= -180.0, lat_lon_array <= 360.0),
+        )
     elif coordinate.is_latitude():
-        valid_indices = np.where(
-            np.logical_and(
-                is_not_fill,
-                np.logical_and(coordinate_row_col >= -90.0, coordinate_row_col <= 90.0),
-            )
-        )[0]
+        valid_indices = np.logical_and(
+            is_not_fill,
+            np.logical_and(lat_lon_array >= -90.0, lat_lon_array <= 90.0),
+        )
     else:
         valid_indices = np.empty((0, 0))
 
-    return valid_indices
+    return valid_indices  # throw an exception
+
+
+def get_row_col_geo_grid_points(
+    lat_arr: np.ndarray,
+    lon_arr: np.ndarray,
+    latitude_coordinate: VariableFromDmr,
+    longitude_coordinate: VariableFromDmr,
+) -> tuple[dict, dict]:
+    """
+    This method is used to return two valid lat lon points from a 2D
+    coordinate dataset. It gets the row and column of the latitude and longitude
+    arrays to get two valid points. This does a check for fill values and
+    This method does not go down to the next row and col. if the selected row and
+    column all have fills, it will raise an exception in those cases.
+    """
```

Comment:
I'm a bit puzzled about this method, and the comment does not really help.  Note 3 line appears to be an incomplete sentence, and the following sentences about fill and progression don't seem to reflect the actual behavior.  The data structure returned seems a bit odd as well - I'm guessing some upstream dependency, but I'm wondering if a simpler structure would do.

Reply:
will update it. It basically returns the lat/lon coordinates based on the valid indices that were determined in the other functions

Reply:
updated - https://github.com/nasa/harmony-opendap-subsetter/pull/18/commits/f0e283a0211293a5f2f891db23af9bbd2ec6c4e1

Reply:
I'm not seeing the update?

Reply:
Sorry. . I thought I updated it. But the contents got duplicated in tthe calling method.. when I refactored..
and maybe I could remove it..  or update the calling method..
- create_dimension_array_from_coordinates(..

Reply:
Just noting that I also don't see the updates for this function's documentation string. Looking at the commit that is referred to above, it's primarily updating the description for the `get_1d_dim_array_data_from_dimvalues` function.

Reply:
updated - https://github.com/nasa/harmony-opendap-subsetter/pull/18/commits/1c8751f9df41672b1414d98f8d0ae3af56f2a362